ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -17,15 +17,39 @@ open Datamodel_types
 open CommonFunctions
 module Types = Datamodel_utils.Types
 
+type fields = string * field list
+
 let templates_dir = "templates"
 
 let ( // ) = Filename.concat
 
+let special_words =
```

Comment:
Make this more efficient: 

```
module StringSet = Set.Make(String) (* already defined below *)

let reserved = 
  ["foo"; "bar" ] |> StringSet.of_list

let is_reserved word = StringSet.mem word reserved
```

* Could also use a `Hashtbl` for this
* The words in this list are acronyms; so could call this `acronyms` and `is_acronym`.